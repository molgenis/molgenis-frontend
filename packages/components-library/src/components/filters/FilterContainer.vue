<template>
  <div
    class="filter-container"
    @mouseup="drag=false"
  >
    <b-collapse
      id="mobile-button-toggle"
      :visible="doCollapse"
    >
      <button
        class="btn w-100 my-2 btn-outline-secondary"
        @click="mobileToggle=!mobileToggle"
      >
        {{ mobileToggle?'Hide filters':'Show filters' }}
      </button>
    </b-collapse>

    <b-collapse
      id="mobile-toggle"
      :visible="!doCollapse || mobileToggle"
    >
      <draggable
        v-model="filtersToShow"
        handle=".drag-handle"
        :class="{'dragdrop': doDragDrop, 'dragging': drag}"
        :disabled="!doDragDrop"
        @choose="drag=true"
        @end="drag=false"
        @input="selectionUpdate"
      >
        <transition-group>
          <filter-card
            v-for="filter in listOfVisibleFilters"
            :key="filter.name"
            v-bind="filter"
            :can-remove="canEdit"
            @removeFilter="removeFilter(filter.name)"
          >
            <component
              :is="filter.type"
              :name="filter.name"
              :value="value[filter.name]"
              v-bind="filter"
              @input="value => selectionChange(filter.name, value)"
            />
          </filter-card>
        </transition-group>
      </draggable>
      <change-filters
        v-if="canEdit && filters.length > 0"
        v-model="filtersToShow"
        :filters="filters"
        @input="selectionUpdate"
      />
    </b-collapse>
  </div>
</template>

<script>
import ChangeFilters from '@/components/filters/ChangeFilters.vue'
import FilterCard from '@/components/filters/FilterCard.vue'
import * as components from '../components/filters'
import draggable from 'vuedraggable'

export default {
  name: 'FilterContainer',
  components: { ChangeFilters, draggable, FilterCard, ...components },
  props: {
    /**
     * List of filter objects
     */
    filters: {
      type: Array,
      required: true
    },
    /**
     * Active filter values
     */
    value: {
      type: Object,
      default: () => ({})
    },
    /**
     * Arrays of filter names to show
    */
    filtersShown: {
      type: Array,
      required: false,
      default: () => []
    },
    /**
     * boolean indicating if the user can change the filters shown
    */
    canEdit: {
      type: Boolean,
      required: false,
      default: () => false
    }
  },
  data () {
    return {
      filtersToShow: this.filtersShown,
      filterToAdd: null,
      drag: false,
      width: 0,
      mobileToggle: false
    }
  },
  computed: {
    doCollapse () {
      // Bootstrap's mobile collapse width
      return this.width <= 576
    },
    doDragDrop () {
      return this.canEdit && !this.doCollapse
    },
    listOfVisibleFilters () {
      return this.filtersToShow.map(id => this.filters.find(filter => filter.name === id))
        .filter(item => item !== undefined)
        .filter(item => item.type !== 'compound-title')
    }
  },
  created () {
    window.addEventListener('resize', this.handleResize)
    this.handleResize()
  },
  destroyed () {
    window.removeEventListener('resize', this.handleResize)
  },
  methods: {
    handleResize () {
      this.width = window.innerWidth
    },
    removeFilter (name) {
      this.filtersToShow = this.filtersToShow.filter(filter => name !== filter)
      this.$emit('update', this.filtersToShow)

      const selections = { ...this.value }
      delete selections[name]
      this.$emit('input', selections)
    },
    selectionChange (name, value) {
      this.$emit('input', { ...this.value, [name]: value })
    },
    selectionUpdate () {
      this.$emit('update', this.filtersToShow)
    }
  }
}
</script>

<style>
  .dragdrop .drag-handle {
    cursor: grab;
  }

  .dragdrop.dragging .drag-handle {
    cursor: grabbing;
  }
</style>

<docs>
Container that holds list of filters

## Usage
```jsx
const selections = { search: 'test'}
const filtersShown = ['datetime', 'disease', 'age']
const updateState = function (newState) { this.filtersShown = newState }
filters: [
        {
          name: 'name',
          label: 'Label',
          description: 'description',
          collapsed: false
        },
        {
          name: 'datetime',
          label: 'Datetime',
          collapsed: false,
          max: null,
          min: null,
          opens: 'right',
          range: true,
          time: false,
          type: 'date-time-filter',
          value: ['01/03/2037', '01/03/2038']
        },
        {
          name: 'age',
          label: 'Age',
          collapsed: false,
          min: -10,
          max: 10,
          step: 0.01,
          type: 'range-filter'
        },
        {
          name: 'disease', // attribute name
          label: 'Filter with multiple options',
          collapsed: false,
          options: mockData.multifilterOptions,
          initialDisplayItems: 5,
          type: 'multi-filter'
        },
        {
          name: 'compound-id',
          label: 'compound',
          description: 'Compound description of a compound',
          type: 'compound-title',
          collapsable: false
        },
        {
          name: 'search',
          label: 'Search',
          description: 'search by string',
          placeholder: 'placeholder',
          type: 'string-filter',
          collapsable: false,
          compound: 'compound-id'
        },
        {
          name: 'color',
          label: 'Color',
          collapsed: false,
          bulkOperation: true,
          options: () => {
            return new Promise(
              function (resolve) {
                resolve(mockData.checkboxOptions)
              })
          },
          type: 'checkbox-filter',
          compound: 'compound-id'
        },
        {
          name: 'number',
          label: 'Number',
          collapsed: false,
          type: 'number-filter',
          compound: 'compound-id'
        },
        {
          name: 'name1',
          label: 'Name',
          description: 'Name of object',
          type: 'string-filter',
          collapsed: false
        },
        {
          name: 'string',
          label: 'String',
          description: 'search by string',
          placeholder: 'placeholder',
          type: 'string-filter',
          collapsable: true,
          collapsed: false
        },
        {
          name: 'checkbox',
          label: 'Checkbox',
          collapsed: false,
          bulkOperation: true,
          maxVisibleOptions: 1,
          options: () => {
            return new Promise(
              function (resolve) {
                resolve(mockData.checkboxOptions)
              })
          },
          type: 'checkbox-filter'
        },
        {
          name: 'checkbox-options',
          label: 'Checkbox lots of options',
          collapsed: false,
          bulkOperation: true,
          maxVisibleOptions: 1,
          options: () => {
            return new Promise(
              function (resolve) {
                resolve(mockData.checkboxLotsOptions)
              })
          },
          type: 'checkbox-filter'
        },
        {
          name: 'long-name',
          label: 'Way too long name to really fit in the user interface',
          collapsed: true,
          bulkOperation: true,
          options: () => {
            return new Promise(
              function (resolve) {
                resolve(mockData.boolCheckboxes)
              })
          },
          type: 'checkbox-filter'
        }
      ]
<FilterContainer
          v-model="selections"
          v-bind:filters="filters"
          v-bind:filters-shown="filtersShown"
          v-bind:can-edit="true"
          v-bind:update="updateState">
</FilterContainer>
<div>model: {{selections}}</div>
```
</docs>
