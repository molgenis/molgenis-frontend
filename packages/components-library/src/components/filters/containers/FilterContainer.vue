<template>
  <div
    class="filter-container"
    @mouseup="drag=false"
  >
    <b-collapse
      id="mobile-button-toggle"
      :visible="doCollapse"
    >
      <button
        class="btn w-100 my-2 btn-outline-secondary"
        @click="mobileToggle=!mobileToggle"
      >
        {{ mobileToggle?'Hide filters':'Show filters' }}
      </button>
    </b-collapse>

    <b-collapse
      id="mobile-toggle"
      :visible="!doCollapse || mobileToggle"
    >
      <draggable
        v-model="filtersToShow"
        handle=".drag-handle"
        :class="{'dragdrop': doDragDrop, 'dragging': drag}"
        :disabled="!doDragDrop"
        @choose="drag=true"
        @end="drag=false"
        @input="selectionUpdate"
      >
        <transition-group>
          <filter-card
            v-for="filter in listOfVisibleFilters"
            :key="filter.name"
            v-bind="filter"
            :can-remove="canEdit"
            @removeFilter="removeFilter(filter.name)"
          >
            <component
              :is="filter.type"
              :name="filter.name"
              :value="value[filter.name]"
              v-bind="filter"
              @input="value => selectionChange(filter.name, value)"
            />
          </filter-card>
        </transition-group>
      </draggable>
      <change-filters
        v-if="canEdit && filters.length > 0"
        v-model="filtersToShow"
        :filters="filters"
        @input="selectionUpdate"
      />
    </b-collapse>
  </div>
</template>

<script>
import ChangeFilters from '../ChangeFilters.vue'
import FilterCard from './FilterCard.vue'
import * as components from '../index'
import draggable from 'vuedraggable'

export default {
  name: 'FilterContainer',
  components: { ChangeFilters, draggable, FilterCard, ...components },
  props: {
    filters: {
      type: Array,
      required: true
    },
    value: {
      type: Object,
      default: () => ({})
    },
    filtersShown: {
      type: Array,
      required: false,
      default: () => []
    },
    canEdit: {
      type: Boolean,
      required: false,
      default: () => false
    }
  },
  data () {
    return {
      filtersToShow: this.filtersShown,
      filterToAdd: null,
      drag: false,
      width: 0,
      mobileToggle: false
    }
  },
  computed: {
    doCollapse () {
      // Bootstrap's mobile collapse width
      return this.width <= 576
    },
    doDragDrop () {
      return this.canEdit && !this.doCollapse
    },
    listOfVisibleFilters () {
      return this.filtersToShow.map(id => this.filters.find(filter => filter.name === id))
        .filter(item => item !== undefined)
        .filter(item => item.type !== 'compound-title')
    }
  },
  created () {
    window.addEventListener('resize', this.handleResize)
    this.handleResize()
  },
  destroyed () {
    window.removeEventListener('resize', this.handleResize)
  },
  methods: {
    handleResize () {
      this.width = window.innerWidth
    },
    removeFilter (name) {
      this.filtersToShow = this.filtersToShow.filter(filter => name !== filter)
      this.$emit('update', this.filtersToShow)

      const selections = { ...this.value }
      delete selections[name]
      this.$emit('input', selections)
    },
    selectionChange (name, value) {
      this.$emit('input', { ...this.value, [name]: value })
    },
    selectionUpdate () {
      this.$emit('update', this.filtersToShow)
    }
  }
}
</script>

<style>
  .dragdrop .drag-handle {
    cursor: grab;
  }

  .dragdrop.dragging .drag-handle {
    cursor: grabbing;
  }
</style>

<docs>
Card containing filter
### Usage
```jsx
const onRemoveFilter = () => alert('remove me please')

const model = {
    search: 'test',
}

const filtersShown = ['datetime']

<div class="row">
  <div class="col-4">
    <FilterContainer
        v-model="model"
        :filters="filtersMocks"
        :filters-shown="filtersShown"
        :can-edit="true"
        @update="(newState) => filtersShown = newState"
    />
  </div>
  <div>{{model}}</div>
</div>
```
