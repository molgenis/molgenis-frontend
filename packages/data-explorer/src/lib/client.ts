import build from './clientFactory'
import store from '@/store/store'
import { AxiosInstance } from 'axios'

export const errorReponse = (error: any) => {
  let message = error.message
  if (error.response && error.response.data && error.response.data.detail) {
    message = error.response.data.detail
  }
  if (error.response && error.response.status === 401) {
    if (!store.getters.isUserAuthenticated) {
      window.location.href = '/login'
    }
  }
  // Set timeout to '0' to keep visible until user dismisses the error notification
  store.commit('addToast', { message, type: 'danger', timeout: 0 })
  return Promise.reject(error)
}

// Create the default axios client the data-explorer will use
const client:AxiosInstance = build({
  responseErrorInterceptor: errorReponse
})

export default client
