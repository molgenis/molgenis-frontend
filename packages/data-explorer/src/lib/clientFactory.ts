/* eslint-disable no-console */
import axios, { AxiosInstance, AxiosResponse } from 'axios'
import { ClientSettings } from '@/types/ClientSettings'

export const defaultResponse = function (response: AxiosResponse<any>) {
  return response
}

export const defaultErrorReponse = function (error: any) {
  if (console && console.error) {
    console.error(`${error.errorCode}: ${error.title}`)
    console.error(`${error.detail}`)
  }
  return Promise.reject(error)
}

export const defaultStatusValidation = function (status: number) {
  return status >= 200 && status < 300
}

export function build (settings?: ClientSettings):AxiosInstance {
  const client:AxiosInstance = axios.create({
    baseURL: settings && settings.baseURL ? settings.baseURL : '/',
    timeout: settings && settings.timeout ? settings.timeout : 5000,
    validateStatus: settings && settings.validateStatus ? settings.validateStatus : defaultStatusValidation
  })

  client.interceptors.response.use(
    settings && settings.responseInterceptor ? settings.responseInterceptor : defaultResponse,
    settings && settings.responseErrorInterceptor ? settings.responseErrorInterceptor : defaultErrorReponse
  )

  return client
}

export default build
