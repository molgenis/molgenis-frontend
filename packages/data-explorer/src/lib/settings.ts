import Axios from 'axios'
import store from '../store/store'

function getSettingsUrl () {
  return `${window.location.href.replace(window.location.hash, '').replace('index.html', '')}config/settings.xlsx`
}

function setSettingsFormData (fileContents: string) {
  let formData = new FormData()
  let dataAsBlob = new Blob([fileContents], { type: 'application/xml' })
  formData.append('file', dataAsBlob, 'settings.xlsx')
  formData.append('action', 'ADD_UPDATE_EXISTING')
  formData.append('notify', 'false')
  return formData
}

function settingsByUrlFormdata () {
  return `/plugin/importwizard/importByUrl?notify=true&entityTypeId=demo&url=${encodeURIComponent(getSettingsUrl())}`
}

const BootstrapExplorer = async () => {
  let versionNumberRequest = await Axios.get('/config/version.txt')
  const version = versionNumberRequest.data
  // check for table
  // if not then check user rights
  // const fileRequest = await Axios.get('/config/settings.xlsx')
  // const settingsFormData = setSettingsFormData(fileRequest.data)
  // console.log(fileRequest)
  const t = settingsByUrlFormdata()
  let x = await Axios.post(t)

  // /plugin/importwizard/importByUrl?notify=false&entityTypeId=demo&url=http://localhost:8080/config/settings.xlsx
  if (store.getters.hasEditRights) { }

  // Axios.get('/config/settingstable.xlsx').then(t => console.log(t))
  let isSettingsTableAvailable = Axios.get('/api/metadata/root_hospital_patients?flattenAttributes=true').then(t =>
    console.log(t)
  )
}

export default BootstrapExplorer
