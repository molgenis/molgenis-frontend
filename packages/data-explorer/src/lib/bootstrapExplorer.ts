import client from './client'
import store from '../store/store'
import applicationSettings from './applicationSettings'

const packageEndpoint = 'api/data/sys_md_Package'
const metaDataEndpoint = 'api/metadata'

async function createSettings () {
  if (!store.getters.hasEditRights) {
    return {
      type: 'danger',
      message: 'Please login as administrator to initialize the application'
    }
  }

  try {
    await client.post(packageEndpoint, applicationSettings.packageSettings)
  } catch (error) {
    // if the settings table already exists, axios gives error and stops executing.
  }

  await client.post(metaDataEndpoint, applicationSettings.entitySettings)
  return {
    type: 'success',
    message: 'The application has been succesfully initialized!'
  }
}

const bootstrapExplorer = async (): Promise<any | undefined> => {
  try {
    await client.get('/api/data/app_set_DataExplorerEntitySettings')
  } catch (error) {
    store.commit('setToast', { type: 'warning', message: 'Bootstrapping Data explorer' })
    const status = await createSettings()
    store.commit('setToast', { type: status.type, message: status.message })
  }
}

export default bootstrapExplorer
