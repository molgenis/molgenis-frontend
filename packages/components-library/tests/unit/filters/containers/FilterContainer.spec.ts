import { mount } from '@vue/test-utils'
import FilterContainer from '@/components/filters/containers/FilterContainer.vue'
import { filters } from '../../../lib/mocks'
import { localVue } from '../../../lib/helpers'

describe('FilterContainer.vue', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = mount(FilterContainer, {
      localVue: localVue(),
      propsData: {
        value: { checkbox: ['red'], string: 'blah' },
        filters,
        filtersShown: ['string', 'checkbox']
      }
    })
  })

  it('consolidates all filter output and sends them via input event', async () => {
    await wrapper.find('input[name="string"]').setValue('test')
    await wrapper.find('input[value="red"]').trigger('click')

    expect(wrapper.emitted().input).toEqual([
      [{ checkbox: ['red'], string: 'test' }],
      [{ checkbox: [], string: 'blah' }]
    ])
  })

  it('cannot remove filters by default', () => {
    expect(wrapper.find('.remove-button').exists()).toBe(false)
  })

  it('can remove filters when editable', async () => {
    await wrapper.setProps({ canEdit: true })
    await wrapper.find('.remove-button').trigger('click')
    expect(wrapper.emitted().update).toEqual([[['checkbox']]])
    expect(wrapper.emitted().input).toEqual([[{ checkbox: ['red'] }]])
  })

  it('can add filters when editable', async () => {
    await wrapper.setProps({ canEdit: true })
    expect(wrapper.find('button.dropdown-toggle')).toBeTruthy()
  })

  it('shows the filters in the correct order ( based on filtersToShow ) ', async () => {
    expect((wrapper.vm).listOfVisibleFilters[0].label).toBe('String')
    expect((wrapper.vm).listOfVisibleFilters[1].label).toBe('Checkbox')
    await wrapper.setData({ filtersToShow: ['checkbox', 'string'] }) // flip order
    expect((wrapper.vm).listOfVisibleFilters[0].label).toBe('Checkbox')
    expect((wrapper.vm).listOfVisibleFilters[1].label).toBe('String')
  })

  it('emits the filtersToShow when selection is changed', async () => {
    await wrapper.setProps({ canEdit: true })
    expect(wrapper.findAll('.card-header').length).toEqual(2)
    await wrapper.find('.change-filters button.btn').trigger('click')
    await wrapper.find('.change-filters ul form input[type=checkbox').trigger('click')
    expect(wrapper.findAll('.card-header').length).toEqual(3)
  })

  it('by default uses a dropdown to select active filters', async () => {
    await wrapper.setProps({ canEdit: true })
    await wrapper.find('.dropdown-toggle').trigger('click')
    expect(wrapper.find('.dropdown-menu')).toBeDefined()
  })

  it('uses a modal to select active filters if dialogStyle is set to modal', async () => {
    await wrapper.setProps({ canEdit: true, dialogStyle: 'modal' })
    await wrapper.find('.btn.btn-block.btn-primary').trigger('click')
    expect(wrapper.find('.modal-dialog')).toBeDefined()
  })
})
