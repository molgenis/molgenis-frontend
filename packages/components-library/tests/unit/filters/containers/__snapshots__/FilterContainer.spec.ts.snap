// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterContainer.vue matches the snapshot 1`] = `
<div
  class="filter-container"
>
  <transition-stub
    css="true"
    enteractiveclass="collapsing"
    enterclass=""
    entertoclass="collapse show"
    leaveactiveclass="collapsing"
    leaveclass="collapse show"
    leavetoclass="collapse"
  >
    <div
      class="collapse"
      id="mobile-button-toggle"
      style="display: none;"
    >
      <button
        class="btn w-100 my-2 btn-outline-secondary"
      >
        
      Show filters
    
      </button>
    </div>
  </transition-stub>
   
  <transition-stub
    css="true"
    enteractiveclass="collapsing"
    enterclass=""
    entertoclass="collapse show"
    leaveactiveclass="collapsing"
    leaveclass="collapse show"
    leavetoclass="collapse"
  >
    <div
      class="collapse show"
      id="mobile-toggle"
    >
      <div
        class=""
      >
        <transition-group-stub>
          <div
            class="card filter-card mb-2"
            placeholder="placeholder"
            type="string-filter"
          >
            <!---->
            <!---->
            <div
              class="card-header drag-handle collapsable"
            >
              <div
                class="title mr-3 px-1"
                title="String"
              >
                <svg
                  aria-hidden="true"
                  class="mr-2 svg-inline--fa fa-caret-right fa-w-6"
                  data-icon="caret-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style="transform: rotate(90deg); transition: transform 0.2s;"
                  viewBox="0 0 192 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class=""
                    d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                    fill="currentColor"
                  />
                </svg>
                
      String
      
                <!---->
              </div>
            </div>
             
            <transition-stub
              css="true"
              enteractiveclass="collapsing"
              enterclass=""
              entertoclass="collapse show"
              leaveactiveclass="collapsing"
              leaveclass="collapse show"
              leavetoclass="collapse"
            >
              <div
                class="collapse show"
                id="string"
              >
                <div
                  class="card-body"
                >
                  <!---->
                  <!---->
                  <fieldset
                    aria-describedby="__BVID__13__BV_description_"
                    class="form-group"
                    id="__BVID__13"
                  >
                    <!---->
                    <div
                      class="bv-no-focus-ring"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="input-group"
                        collapsable="true"
                        description="search by string"
                        label="String"
                        role="group"
                        type="string-filter"
                      >
                        <!---->
                        <input
                          class="form-control"
                          id="__BVID__15"
                          name="string"
                          placeholder="placeholder"
                          type="text"
                        />
                         
                        <div
                          class="input-group-append"
                        >
                          <button
                            class="btn btn-outline-secondary"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="svg-inline--fa fa-times fa-w-11"
                              data-icon="times"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              viewBox="0 0 352 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                class=""
                                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                fill="currentColor"
                              />
                            </svg>
                          </button>
                        </div>
                        <!---->
                      </div>
                      <!---->
                      <!---->
                      <small
                        class="form-text text-muted"
                        id="__BVID__13__BV_description_"
                        tabindex="-1"
                      >
                        search by string
                      </small>
                    </div>
                  </fieldset>
                </div>
              </div>
            </transition-stub>
            <!---->
            <!---->
          </div>
          <div
            bulkoperation="true"
            class="card filter-card mb-2"
            maxvisibleoptions="1"
            options="() => {
    return new Promise(function (resolve) {
      resolve(colorOptions);
    });
  }"
            type="checkbox-filter"
          >
            <!---->
            <!---->
            <div
              class="card-header drag-handle collapsable"
            >
              <div
                class="title mr-3 px-1"
                title="Checkbox"
              >
                <svg
                  aria-hidden="true"
                  class="mr-2 svg-inline--fa fa-caret-right fa-w-6"
                  data-icon="caret-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style="transform: rotate(90deg); transition: transform 0.2s;"
                  viewBox="0 0 192 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class=""
                    d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                    fill="currentColor"
                  />
                </svg>
                
      Checkbox
      
                <!---->
              </div>
            </div>
             
            <transition-stub
              css="true"
              enteractiveclass="collapsing"
              enterclass=""
              entertoclass="collapse show"
              leaveactiveclass="collapsing"
              leaveclass="collapse show"
              leavetoclass="collapse"
            >
              <div
                class="collapse show"
                id="checkbox"
              >
                <div
                  class="card-body"
                >
                  <!---->
                  <!---->
                  <fieldset
                    class="form-group"
                    id="__BVID__19"
                  >
                    <!---->
                    <div
                      class="bv-no-focus-ring"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        label="Checkbox"
                        name="checkbox"
                        type="checkbox-filter"
                      >
                        <div
                          class="bv-no-focus-ring"
                          id="__BVID__21"
                          role="group"
                          tabindex="-1"
                        >
                          <div
                            class="custom-control custom-checkbox"
                          >
                            <input
                              class="custom-control-input"
                              id="__BVID__21_BV_option_0"
                              name="__BVID__21"
                              type="checkbox"
                              value="red"
                            />
                            <label
                              class="custom-control-label"
                              for="__BVID__21_BV_option_0"
                            >
                              <span>
                                Red
                              </span>
                            </label>
                          </div>
                        </div>
                         
                        <span>
                          <a
                            class="toggle-slice card-link"
                            href="#"
                            target="_self"
                          >
                            
      Show 6 more
    
                          </a>
                           
                          <a
                            class="toggle-select card-link"
                            href="#"
                            target="_self"
                          >
                            
      Deselect all
    
                          </a>
                        </span>
                      </div>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </fieldset>
                </div>
              </div>
            </transition-stub>
            <!---->
            <!---->
          </div>
        </transition-group-stub>
      </div>
       
      <!---->
    </div>
  </transition-stub>
</div>
`;
